src/support/z_cowboy_log_stream_handler.erl

```diff
@@ -0,0 +1,39 @@
+-module(z_cowboy_log_stream_handler).
+-author("David de Boer <david@ddeboer.nl>").
+
+-behavior(cowboy_stream).
+
+-export([
+    init/3,
+    data/4,
+    info/3,
+    terminate/3
+]).
+
+-type state() :: any().
+
+-spec init(cowboy_stream:streamid(), cowboy_req:req(), cowboy:opts())
+	-> {cowboy_stream:commands(), {module(), state()} | undefined}.
+init(StreamID, Req, Opts) ->
+    z:debug_msg(?MODULE, ?LINE, Req),   %% Map with request headers and properties
+    cowboy_stream:init(StreamID, Req, Opts).
+
+-spec data(cowboy_stream:streamid(), cowboy_stream:fin(), binary(), {Handler, State} | undefined)
+	-> {cowboy_stream:commands(), {Handler, State} | undefined}
+	when Handler::module(), State::state().
+data(StreamID, IsFin, Data, {Handler, State0}) ->
+    z:debug_msg(?MODULE, ?LINE, Data),  %% Response body
+    cowboy_stream:data(StreamID, IsFin, Data, {Handler, State0}).
```

Comment:
I... don't know Zotonic enough to know what it means. What I do know, however, is that you probably will need to parse the host in the stream handler, independently of the routing of requests (and perhaps cache that info in the environment ultimately given to middlewares).